File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/ShaderCode.java
package org.saar.lwjgl.opengl.shaders;

public class ShaderCode {

    private final String code;

    private ShaderCode(String code) {
        this.code = code;
    }

    public static ShaderCode define(String name, String value) {
        return new ShaderCode("#define " + name + " " + value);
    }

    public static ShaderCode loadSource(String file) throws Exception {
        final String code = ShaderCodeLoader.loadSource(file);
        return new ShaderCode(code);
    }

    public String getCode() {
        return this.code;
    }
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shader/ShaderCode.kt
package org.saar.lwjgl.opengl.shader

import org.saar.rhi.shader.ShaderModuleLoader

class ShaderCode private constructor(val code: String) {

    companion object {
        fun define(name: String, value: String): ShaderCode {
            return ShaderCode("#define $name $value")
        }

        @Throws(Exception::class)
        fun loadSource(file: String): ShaderCode {
            val code = ShaderModuleLoader.loadSource(file)

            return ShaderCode(code)
        }
    }
}
